### PR TITLE
st2 [2808][ADD] stock_return_picking_refund

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[2808](https://www.quartile.co/web#view_type=form&model=project.task&id=2808&active_id=2808&menu_id=)

### CODE REVIEW

File: stock_picking_reverse_auto_check_refund/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This module does the following:
+- Default check "To Refund (update SO/PO)" in the reverse transfer of inventory transfer.
```
Comment (yostashiro) [inline stock_picking_reverse_auto_check_refund/readme/DESCRIPTION.rst]:
```suggestion

- Default check "To Refund (update SO/PO)" in the reverse transfer of inventory transfer.
```
The layout breaks without a blank line.

File: stock_picking_reverse_auto_check_refund/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2019-2020 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock Picking Reverse Auto Check Refund",
+    "summary": "",
```
Comment (yostashiro) [inline stock_picking_reverse_auto_check_refund/__manifest__.py]:
```suggestion
```

File: stock_picking_reverse_auto_check_refund/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2019-2020 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (yostashiro) [inline stock_picking_reverse_auto_check_refund/__manifest__.py]:
```suggestion
# Copyright 2022 Quartile Limited
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
```

File: stock_picking_reverse_auto_check_refund/models/stock_return_picking.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from odoo import models, fields, api
```
Comment (yostashiro) [inline stock_picking_reverse_auto_check_refund/models/stock_return_picking.py]:
```suggestion
# Copyright 2022 Quartile Limited
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).

from odoo import api, fields, models
```

Please run pre-commit.

File: stock_picking_reverse_auto_check_refund/models/stock_return_picking.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+class StockReturnPicking(models.TransientModel):
+    _inherit = "stock.return.picking"
+
+    @api.model
+    def default_get(self, default_fields):
+        res = super(StockReturnPicking, self).default_get(default_fields)
+        for i, k, vals in res.get('product_return_moves', []):
+            vals.update({'to_refund': True})
+        return res
```
Comment (yostashiro) [inline stock_picking_reverse_auto_check_refund/models/stock_return_picking.py]: Shouldn't it suffice if we just assign the default value to this field https://github.com/odoo/odoo/blob/12.0/addons/stock_account/models/stock.py#L775?

**Review by yostashiro — CHANGES_REQUESTED** [review]
Can we change the module name to stock_return_picking_refund?

File: stock_picking_reverse_auto_check_refund/models/stock_return_picking.py (hunk 5 above)
Comment (AungKoKoLin1997) [reply stock_picking_reverse_auto_check_refund/models/stock_return_picking.py]: We can also assign the default value to that field. But I think we assign default value by inheriting function is better way than overriding field in this case(just my POV).

File: stock_picking_reverse_auto_check_refund/models/stock_return_picking.py (hunk 4 above)
Comment (AungKoKoLin1997) [reply stock_picking_reverse_auto_check_refund/models/stock_return_picking.py]: I run the pre-commit and there is a [PR](https://github.com/qrtl/hls-custom/pull/191) for fixing pre-commit issues.

**Review by yostashiro — CHANGES_REQUESTED** [review]
Please do not include the pre-commit fixes of other modules, as I mentioned.

Rename the file from stock_return_picking.py to stock_return_picking_line.py.

**Review by yostashiro — APPROVED** [review]
@AungKoKoLin1997 Please create another task (and PR to follow up on it) for the pre-commit issue.

**Review by kanda999 — APPROVED** [review]
I have fixed error in __init__ file.
Function is working as expected.

**Pull request merged**